lib/engine/stock_market.rb

```diff
@@ -5,12 +5,13 @@
 
 module Engine
   class StockMarket
-    attr_reader :market, :par_prices, :has_close_cell, :zigzag
+    attr_reader :market, :par_prices, :has_close_cell, :zigzag, :ledge_movement
 
-    def initialize(market, unlimited_types, multiple_buy_types: [], zigzag: nil)
+    def initialize(market, unlimited_types, multiple_buy_types: [], zigzag: nil, ledge_movement: nil)
       @par_prices = []
       @has_close_cell = false
       @zigzag = zigzag
+      @ledge_movement = ledge_movement
```

Comment:
We don't need to store ledge_movement or zigzag, for that matter. They are only used to create the stock_movement class during the constructor.

Reply:
Let me know if you want to make this change or leave as is.

Reply:
I'll change it

Reply:
Turns out the zigzag attribute is still needed. Both game/base and the stock market UI need it.